### PR TITLE
refactor: replace f64::MIN with values closer to 0

### PR DESCRIPTION
**Pull request by davidwalschots**
## Summary of Changes
As part of #4912 I introduced `f64::MIN` as a value indicating "no value". As `f64::MIN` is a very large negative number not all other applications (such as yourcontrols) handle it well. I've now replaced `f64::MIN` with a new mechanism for reading and writing, using Rust traits.

The file in which I originally started writing `f64::MIN` is found in `src/systems/systems/src/apu/electronic_control_box.rs`. `src/systems/systems/src/simulation/mod.rs` contains the new traits and implementations.

## Additional context

Discord username (if different from GitHub): SjotgunSjonnie#9623

## Testing instructions
A basic test of ELEC, HYD, and APU functionality will suffice. The change affects the reading and writing of variables to the simulator. Thus look for any values being unexpected on ECAM pages, etc.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: docs/a320-simvars.md [hunk 1]
```diff
@@ -360,30 +360,30 @@
 - A32NX_APU_EGT_CAUTION
     - Celsius
     - The APU's exhaust gas temperature caution level, to be indicated in amber in the cockpit,
-      when f64::MIN/-Number.MAX_VALUE the ECB isn't supplying information, for example due to being unpowered.
+      when < -273.15 the ECB isn't supplying information, for example due to being unpowered.
```
Comment (tracernz) [inline docs/a320-simvars.md:363]: Well, we are in trouble if the ECB gets that cold. At least we won't have to worry about it.

**Review by tracernz — APPROVED** [review]

File: docs/a320-simvars.md (hunk 1 above)
Comment (davidwalschots) [reply docs/a320-simvars.md:363]: I think the whole universe would have collapsed by that point. 😀

**Comment by TheSquirtingPantin** [comment]
Quality Assurance Tester

Discord: Remi#8871
Object of testing: #4998
Tier of Testing: 1
Date: 28/05/2021

Testing Process:
1) CD at KIAD
2) APU Start and visual inspection of APU ECAM page
3) ENG 2 start and visual inspection of HYD page
4) Inspection of ELEC page

Negatives: Post ENG 2 start, the PTU transfer is not occuring. No transfer noticed on the HYD RtoL variable (model behavior). Not sure this bug is being introduced by this PR.
No other issues with APU HYD and ELEC pages found. 

Testing Results: To Be Improved. 

Conclusions: Problem with PTU indication during engine start.  
Media: N/A

**Comment by marcman86** [comment]
Quality Assurance Tester Report

Discord:marcman86#4907
Object of testing: #4998
Tier of Testing: 1
Date: 04/06/2021

Testing Process: 
- Spawned C&D at EDDP
- Start up
- Start APU, looked at APU and ELEC page
- Start engine 1, looked at HYD and ELEC page
- Start engine 2, looked at HYD and ELEC page
- Turn off APU, looked at ELEC and ELEC page

Negatives: -
Testing Results: Passed
Conclusions: Works
Media: -
Hope this helps :-)

**Pull request merged**